src/mbgl/map/map.cpp

```diff
@@ -67,6 +67,7 @@ class Map::Impl : public style::Observer {
 
     std::string styleURL;
     std::string styleJSON;
+    bool styleMutated = false;
```

Comment:
I wanted to do it that way, but the issue is that we want some style mutations to be excluded from this logic, specifically those that `AnnotationManager::updateStyle` performs. Adding annotations should not cancel the initial revalidation. (I'll add a test for this.)